### DISCUSSION TITLE
Shellcheck reporting no errors in output, but also doing ... nothing

### DISCUSSION THREAD

**Discussion by ELLIOTTCABLE** [timonwong/vscode-shellcheck#907] [Q&A] [answered]
In a VScode Remote workspace, I've got Shellcheck installed and enabled; but the output pane shows nothing other than this:

```
[2023-01-30T19:11:09.888Z] [INFO] shellcheck (bundled) version: 0.9.0
```

Running the `shellcheck.runLint` command has … no effect. No error message from VScode, nothing in the Output window; but also no errors are reported in the script. (I've intentionally added errors in an attempt to trigger … some? sort of response from ShellCheck.)

Running `shellcheck` from the command-line reports plenty of errors.

What can I do to debug this? I can't think of a next step. /=

(**Edit:** Nothing in `settings.json` relating to vscode-shellcheck)

**Comment by felipecrs** [comment, marked as answer]
Can you please post your vscode-shellcheck settings?

**Reply by ELLIOTTCABLE** [reply]
Updated OP; no settings configured whatsoever.

However, _adding_ the following still has no effect, which is worrying:

```json5
   "[shellscript]": {
      "shellcheck.executablePath": "somethingthatdoesntexist"
   },
```

(As in, no further output in the shellcheck Output pane; no error-popup in the bottom-right in VScode, etc.)

**Reply by felipecrs** [reply]
It should not be under `[shellscript]` by the way.

**Reply by felipecrs** [reply]
> Updated OP; no settings configured whatsoever.

Did you check all settings levels? User, machine/devcontainer/codespaces, workspace...

**Reply by ELLIOTTCABLE** [reply]
Ah, okay, well at least the reason that has no effect is documented:

<img width="689" alt="image" src="https://user-images.githubusercontent.com/200/215582018-6b98aea1-0535-431c-97dc-ff2f36b57383.png">

> This setting cannot be applied in this window. It will be applied when you open a local window.

**Reply by felipecrs** [reply]
Maybe that's why it's not working as well. I don't understand it by the way. Are you running VS Code remote? What kind of remote connection?

**Reply by ELLIOTTCABLE** [reply]
Yeah, we work using [VSCode Remote - SSH](https://code.visualstudio.com/docs/remote/ssh), with all actual development and compilation happening on beefy devboxes. `vscode-shellcheck`, as with many such extensions, automatically installed itself serverside instead of locally:

<img width="383" alt="image" src="https://user-images.githubusercontent.com/200/215583375-1a613683-a72d-415b-a2b7-d72ab99f3171.png">

**Reply by felipecrs** [reply]
Ok. I use Remote SSH the entire day and it works just fine for me.

**Reply by felipecrs** [reply]
What's the architecture of your machine? x86_64? Is your machine Linux?

**Reply by felipecrs** [reply]
By the way, there should not be a need to specify executable path as the binary for shellcheck already comes bundled with the extension.

**Reply by ELLIOTTCABLE** [reply]
Okay, just figured itself out; turns out I wasted your time — for whatever reason, the default configuration doesn't include `.zsh` files? Renaming the file to `.sh` has it working.

I'd _expect_ it to work by default on any file that VScode decides is a `shellscript` filetype; especially if I explicitly set one. I'm assuming it's defaulting to `shellcheck`'s command-line behaviour ... which should probably be overriden in the VScode environment to follow VScode conventions, maybe.

Is this expectation work opening an Issue over? (=

**Reply by felipecrs** [reply]
Shellcheck doesn't support ZSH, does it?

**Reply by felipecrs** [reply]
Well, glad that you found out. :)
